### DISCUSSION TITLE
Adding Compatibility Matrixes to Repositories

### DISCUSSION THREAD

**Discussion by ds-mmaul** [eclipse-tractusx/sig-infra#373] [Ideas]
### Summary

I've been thinking about an idea that I believe could be beneficial -->  the inclusion of compatibility matrixes in each repository.

### Description
Currently, our release guidelines (TRGs) include the provision of crucial information about helm charts. However, I'd like to suggest a more detailed approach by introducing compatibility matrixes directly in the root of each repository. These matrixes would include information about the version of the app itself and services which it depends on and components that are compatible with the current release.

### Example

Here's a sample format for the compatibility matrix from our repo:

| Name of service    | Version | Comments   |
|--------------------|---------|------------|
| postgresql         | 12.1.6  |            |
| edc-postgresql     | 12.1.6  |            |
| pgadmin4           | 1.13.6  |            |
| irs-helm           | 6.9.1   |            |
| tractusx-connector | 0.5.3   |            |
| discovery service  | 0.1.0   |            |
| portal             | ?       |            |
| SD-Factory         | ?       |            |
| wallet             | ?       |            |

which also can bee seen as the whole file in our repo: https://github.com/catenax-ng/tx-traceability-foss/blob/main/COMPATIBILITY_MATRIX.md

### Benefit
Including this information in each repository can help streamline the release process, ensuring that users have clear insights into the compatibility requirements for each release. It promotes transparency and facilitates a smoother experience for both contributors and users.

### Discussion
I'd love to hear your thoughts on this proposal. Are there any additional considerations or modifications you think would improve this idea? I would like to discuss and see if this could be an senseful enhancement to the TRGs.

Looking forward to your feedback!

Best regards,
Martin Maul
From the traceability-foss project

**Comment by evegufy** [comment]
It's a really important topic, so great that you bring it up!
IMO the [official changelog from TX](https://eclipse-tractusx.github.io/CHANGELOG) does already provide such a compatibility overview, as it reflects the versions which were integration tested together.
If possible, I'd like to keep one single source of truth regarding compatibility and enhance the the official changelog if there's some information missing. In order to provide a more widespread knowledge about compatibility, maybe the official changelog should be linked in the README, INSTALL.md or CHANGELOG.md of every repository.

**Comment by Siegfriedk** [comment]
In best case the matrix has also links to successful e2e tests from the upcoming umbrella e2e testing chart.

I would also try to avoid adding patch versions. Should we expect that patch versions are always compatible, i believe.

**Comment by jzbmw** [comment]
I also vote for the topic @evegufy brought up. 
We should definitely keep an eye on single source of truth.

**Comment by ds-mmaul** [comment]
Thanks for the discussion so far! That makes sense to me. 

@evegufy I like the idea of referring to the TX changelog in the repository. It makes the interaction and compatibility of the tx components/repos more visible to a developer, which I think is important. 

I will make a suggestion in form of a pull request, there we can continue discussing the details on where/what to put in the TRGs and how it could be referenced in the repo.

@Siegfriedk  - could you make an example? We could throw the suggestions together and concretize them through a pull request.

**Reply by ds-mmaul** [reply]
PR result: https://github.com/eclipse-tractusx/tractus-x-release/pull/18

**Reply by Siegfriedk** [reply]
Unfortunate i have way to little experience what compatiblities are in tractus-x. I would point to @RolaH1t but he is currently on holiday.

While i suggest things for the official changelog, @RolaH1t is writing it.

We would need a type of 'questions release catalog': Whenever a release happens, we send this to everyone and than everyone would add dependencies and compatibility matrix. 

If we want to have it for all repos, we should put it into a TRG.

**Reply by Siegfriedk** [reply]
@ds-mmaul @RolaH1t is back from holiday
